consensus/consensus-types/src/executed_block.rs

```diff
@@ -13,12 +13,13 @@ use diem_types::{
     transaction::{Transaction, TransactionStatus},
 };
 use executor_types::StateComputeResult;
+use serde::{Deserialize, Serialize};
 use std::fmt::{Debug, Display, Formatter};
 
 /// ExecutedBlocks are managed in a speculative tree, the committed blocks form a chain. Besides
 /// block data, each executed block also has other derived meta data which could be regenerated from
 /// blocks.
-#[derive(Clone, Eq, PartialEq)]
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize)]
```

Comment:
where do we need to serialize it?

Reply:
In safety-rule, to perform the extension check (commented), we need to give the safety-rule the executed block (or both the block info and state compute result).